client/components/list-details.jsx

```diff
@@ -1,18 +1,26 @@
 import React from 'react';
 import ListItem from './list-item';
+import AppContext from '../lib/app-context';
 
 export default class ListDetails extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
+      token: null,
       listData: null
     };
   }
 
   componentDidMount() {
+    const { token } = this.context.token;
+    this.setState({ token });
```

Comment:
You don't have to store the token in state like this, you can refer to `this.context.token` on-demand.

Reply:
Corrected